### ISSUE TITLE
Contact broken after change on mobile device and sync with DAVx5

### ISSUE THREAD

**Issue by FelixBuehler** [nextcloud/contacts#1286]
### Describe the bug
When I change a contact on my mobile device and then synchronize the changes via DAVx5 to my Nextcloud Contact App, the contact gets the message
""This contact was broken and received a fix. Please review the content and click here to save it."

For now everything seems to be fine (no corrupt data or so), but the error message appears all the time.
Can't exactly tell whether the problem is the contact app or DAVx5 (how to determine?)

In the console is written (yes, twice, so maybe a error where contacts are getting created multiple times?)
```
contacts.js?v=987a53c4-0:1086 The following contact needed a correction that failed: duplicate types t
(anonymous) @ contacts.js?v=987a53c4-0:1086
contacts.js?v=987a53c4-0:1086 The following contact needed a correction that failed: duplicate types t
```
**To Reproduce**
1. Change contact on mobile device
2. Snyc to cloud
3. Open contact in cloud

### Server configuration

**Web server:** Apache 2.4

**Database:** MariaDB 10

**PHP version:** 7.3

**Nextcloud version:** 16.0.5

**Contacts version:** 	3.1.4

### Client configuration
**Browser:** Vivaldi

**Operating system:** Windows 10

**CardDAV-clients:** DAVx5

#### *Web server error log*
```

**Comment by skjnldsv** [comment]
Hello! :)
This is not a bug! Your contacts had duplicate types in it which was automatically cleaned.
Have a look at the content, if everything seems fine, click the button to save it as it

Duplicate types is something like that:
`TEL;TYPE="WORK,WORK":0123456789`, only one WORK is needed here, so we're cleaning it for you !

**Comment by FelixBuehler** [comment]
Well, I created a "test"-contact with only a dummy-number "0123456789". Synchronized it with my mobile device, changed name to "test2", synchronized back again and got the error.
So either my smartphone creates the contacts already bugged, or something else is wrong?

**Comment by skjnldsv** [comment]
@FelixBuehler can you post the contact here?

**Comment by FelixBuehler** [comment]
Sure, find it attached. The message in the contact info however shows "This contact was broken and received a fix. Please review the content and click here to save it." so I#M not quite sure, if the file attached is already a fixed version.
[BAE13DF3-4C59-41AF-A086-5D27D25CA89F.zip](https://github.com/nextcloud/contacts/files/3667559/BAE13DF3-4C59-41AF-A086-5D27D25CA89F.zip)
So i also exported the contact from my phone:
[test1_0.zip](https://github.com/nextcloud/contacts/files/3667579/test1_0.zip)

**Comment by skjnldsv** [comment]
Your test contact is a vcf 2.1, which we don't support, so it's normal that it is failing.

```
BEGIN:VCARD
VERSION:2.1
N:;test1;;;
FN:test1
TEL;HOME;PREF:0123456789
END:VCARD

```

For the other one, I see nothing wrong here, maybe the one you downloaded was after correction? If you click the icon, it get sent to the server, right? If you refresh, do you see the error again?

```
BEGIN:VCARD
VERSION:3.0
PRODID:-//Sabre//Sabre VObject 4.1.6//EN
UID:a5174458-4305-446d-b08f-69328231a41d
FN:test1
N:;test1;;;
TEL;TYPE=home,PREF:0123456789
REV:20190929T100418Z
END:VCARD
```

**Comment by FelixBuehler** [comment]
The 2.1 vcf is the one exported from my phone. 

> maybe the one you downloaded was after correction?
Yeah, like I mentioned above: It looks like the contact gets corrected automatically, I just can click the button to confirm that everything shown is correct. That's probably the cause for the file (v3.0( looking correct.

So in the 2.1 vcf, is the version the only reason for a "corrupt" file? Or do you see any other errors?

**Comment by skjnldsv** [comment]
> Yes, the error shows as long as i don't confirm that everything is correct.

But gone once you clicked it or changed anything on the contact (name, tel, address...) ?


> So in the 2.1 vcf, is the version the only reason for a "corrupt" file? Or do you see any other errors?

A 2.1 contact will be refused by nextcloud directly. So if you see it on nextcloud, it means it have been converted by your contact sync manager (davx5?)

**Comment by FelixBuehler** [comment]
> But gone once you clicked it or changed anything on the contact (name, tel, address...) ?

Yes, both. The message disappears if I either click "confirm" or change anything in the contact via nextcloud.

So I/we are still not sure yet, what causes the error message, right?

**Comment by skjnldsv** [comment]
> So I/we are still not sure yet, what causes the error message, right?

Yes we do, but the vcard you sent is the fixed one, you can try by downloading one that still have the message (so, not fixed yet) and paste it here :)

**Comment by FelixBuehler** [comment]
Like I said, I can't. When i edited something via my phone and sync it, I get the info message
"This contact **was** broken and received a fix. Please review the content and click here to save it."
So even when the message is showing and I click download, I get the file above.

**Comment by skjnldsv** [comment]
AH, then it means there might eb something wrong indeed, let me see :)

**Comment by skjnldsv** [comment]
Nope, tried importing yours, no issue whatsoever :)

We'll try to improve the errors handling anyway. Display better debugging somewhere. But I'd say it's a bit weird, trying to stay non technical for people that don't know anything about vcards and trying to not auto-sync it, I'd say it's a hard task.

Maybe I should automatically sync the fixed vcard without an error message.
Our current auto fixes are simple enough that it won't risk data loss anyway. :)

**Comment by thenickname** [comment]
I stumbled upon this bug today as well. It still seems to be there. I've tried editing a contact with the default Android contacts app as well as the app "Simple Contacts". After syncing through davx5 nextcloud says that the contact is broken and received a fix. While merely annoying and not a major issue, this seems to be a bug nonetheless. Why was this issue closed?

Of course I cannot exactly pinpoint the source of the error, it could well be an issue with davx5 or the contact apps. But since I tried two different apps, I think it's at least worth considering that this is an issue with nextcloud.

Sysinfo: Android 9
